modules/core/src/main/scala/org/tessellation/l0/SnapshotService.scala

```diff
@@ -0,0 +1,59 @@
+package org.tessellation.l0
+
+import cats.Applicative
+import cats.effect.kernel.Async
+import cats.syntax.flatMap._
+import cats.syntax.functor._
+
+import org.tessellation.config.types.SnapshotConfig
+import org.tessellation.dag.block.{BlockStorage, BlockValidator}
+import org.tessellation.schema.height.Height
+
+import eu.timepit.refined.auto._
+import eu.timepit.refined.types.numeric.PosLong
+import io.estatico.newtype.ops._
+
+trait SnapshotService[F[_]] {
+  def createSnapshot(minTipHeight: PosLong): F[DAGSnapshot]
+}
+
+object SnapshotService {
+
+  def make[F[_]: Applicative: Async](
+    config: SnapshotConfig,
+    blockStorage: BlockStorage[F],
+    blockValidator: BlockValidator[F],
+    snapshotStorage: SnapshotStorage[F]
+  ): SnapshotService[F] =
+    new SnapshotService[F] {
+
+      def createSnapshot(minTipHeight: PosLong): F[DAGSnapshot] =
+        for {
+          previousSnapshotHeight <- snapshotStorage.getLastSnapshotHeight
+          nextSnapshotHeight = Height(previousSnapshotHeight.coerce + config.heightInterval)
+
+          // In 1.0: validateSnapshotHeightIntervalCondition
+          //  if (minTipHeight > nextHeight + config.heightDelayInterval)
+
+          acceptedBlocks <- blockStorage.getAccepted
+
+          blocksForNextSnapshot = acceptedBlocks.filter { block =>
+            val blockHeight = block.signed.height.coerce
+            blockHeight > previousSnapshotHeight.coerce && blockHeight <= nextSnapshotHeight.coerce
+          }
+
+          // In 1.0: validateAcceptedCBsSinceSnapshot
+          //  if (blocksForNextSnapshot.nonEmpty) OK
```

Comment:
We shouldn't create an empty snapshot but it shouldn't be checked here as well. Actually, I have a feeling that all the conditions should be checked outside of `createSnapshot` function. WDYT?

Reply:
I think it makes sense. At least for now, until we figure out the triggering mechanism.

Reply:
The problem here is that you can't pull from the storage and expect that pulled blocks will not be and empty `List`/`Set`. It would need to be more like a pure function getting `NEL` or `NES` of blocks as a parameter.

Reply:
But we can't guarantee that it will be NEL/NES. Eventually `Option[NonEmptyList]` 🤔